homeassistant/components/mqtt/sensor.py

```diff
@@ -239,6 +239,12 @@ def _update_state(msg: ReceiveMessage) -> None:
                 self._attr_native_value = new_value
                 return
             try:
+                if (dt_util.parse_datetime(new_value)) is None and (
+                    string_datetime := datetime.fromtimestamp(
+                        float(payload), tz=UTC
+                    ).isoformat()
+                ) is not None:
+                    new_value = string_datetime
```

Comment:
For every non-time sensor this code would need to check on timestamps as well.

Reply:
Sensors with device classes that are numeric will be treated in this part of the code:
https://github.com/home-assistant/core/blob/abda764241e38ffcf8711cbf163d671fd52bb046/homeassistant/components/mqtt/sensor.py#L230-L237

Sensors that do not have a device class or that have the ENUM device class will be treated in this part of the code:
https://github.com/home-assistant/core/blob/abda764241e38ffcf8711cbf163d671fd52bb046/homeassistant/components/mqtt/sensor.py#L238-L240

Therefore, only the sensors with the TIMESTAMP and DATE device classes are left to be treated by continuing the code:
https://github.com/home-assistant/core/blob/abda764241e38ffcf8711cbf163d671fd52bb046/homeassistant/components/mqtt/sensor.py#L241-L259

So I understand that non-time sensors will not pass this code.

If I'm wrong, please let me know...